source/critical.c

```diff
@@ -15,7 +15,9 @@
  * limitations under the License.
  */
 
-#ifndef TARGET_NORDIC
+// This critical section implementation is generic for mbed OS targets,
+// except Nordic ones
+#if defined(TARGET_LIKE_MBED) && !defined(TARGET_NORDIC)
```

Comment:
Not really. This is linked to the POSIX platform, since the previous condition (`#ifndef TARGET_NORDIC`) assumed that the only possibilities are "regular" mbed OS and Nordic platforms. This change acknowledges the presence of another platform (POSIX).